hydra-main/src/main/java/com/addthis/hydra/query/MeshQueryMaster.java

```diff
@@ -126,6 +130,18 @@ public MeshQueryMaster(QueryTracker tracker) throws Exception {
         }
     }
 
+    public AtomicBoolean getIsActive() {
+        return isActive;
+    }
+
+    public boolean deactivateMqMaster(boolean deactivate) {
+        if(deactivate) {
+            // Set mqMaster's isActive field to false
+            return this.isActive.compareAndSet(true, !deactivate);
+        }
+        return this.isActive.compareAndSet(false, true);
```

Comment:
This whole method could be replaced with `this.isActive.compareAndSet(deactivate, !deactivate)`. What are we doing on the client side if compare and set is not successful?